moses/atomese/representation/atomese_representation.cc

```diff
@@ -0,0 +1,467 @@
+ * on its row (i.e Predicate of feature on Node of row).
+ *
+ * @param b builtin value
+ * @param row_num row number
+ * @param label label of the feature
+ * @return
+ */
+Handle mk_boolean_cell_eval(const builtin b,
+                            const int row_num, const string& label)
+{
+	Handle h = createLink(EVALUATION_LINK,
+	                      createNode(PREDICATE_NODE, label),
+	                      mk_row_number_cell(row_num));
+	TruthValuePtr tv(SimpleTruthValue::createTV(builtin_to_bool(b), 1));
+	h->setTruthValue(tv);
+	return h;
+}
+
+/**
+ * Represent a boolean cell as an Execution of the corresponding feature
+ * on its row (i.e Schema of feature on Node of row).
+ *
+ * @param b builtin value
+ * @param row_num row number
+ * @param label label of the feature
+ * @return
+ */
+Handle mk_boolean_cell_exec(const builtin b,
+                            const int row_num, const string& label)
+{
+	return createLink(EXECUTION_LINK,
+	                  createNode(SCHEMA_NODE, label),
+	                  mk_row_number_cell(row_num),
+	                  mk_boolean_cell(b));
+}
+
+struct mk_cell_visitor : public boost::static_visitor<Handle>
+{
+	Handle operator()(const builtin b) const
+	{
+		return mk_boolean_cell(b);
+	}
+
+	Handle operator()(const contin_t contin) const
+	{
+		return mk_real_cell(contin);
+	}
+
+	template<typename T>
+	Handle operator()(const T& t) const
+	{}
+};
+
+struct mk_exec_cell_visitor : public boost::static_visitor<Handle>
+{
+	Handle operator()(const builtin b) const
+	{
+		return mk_boolean_cell_exec(b, row_num, label);
+	}
+
+	Handle operator()(const contin_t contin) const
+	{
+		return mk_real_cell_exec(contin, row_num, label);
+	}
+
+	template<typename T>
+	Handle operator()(const T& t) const
+	{}
+
+	int row_num;
+	string label;
+};
+
+struct mk_cell_seq_visitor : public boost::static_visitor<HandleSeq>
+{
+	HandleSeq operator()(const builtin_seq& b) const
+	{
+		HandleSeq cellSeq;
+		for (builtin cell_value : b)
+			cellSeq.push_back(mk_boolean_cell(cell_value));
+
+		return cellSeq;
+	}
+
+	HandleSeq operator()(const contin_seq& c) const
+	{
+		HandleSeq cellSeq;
+		for (contin_t cell_value : c)
+			cellSeq.push_back(mk_real_cell(cell_value));
+
+		return cellSeq;
+	}
+
+	template<typename T>
+	HandleSeq operator()(const T& t) const
+	{}
+};
+
+struct mk_exec_cell_seq_visitor : public boost::static_visitor<HandleSeq>
+{
+	HandleSeq operator()(const builtin_seq& b)
+	{
+		for (builtin cell_value : b)
+			cellSeq.push_back(mk_boolean_cell_exec(cell_value, row_num,
+			                                       labels[cell_number++]));
+
+		return cellSeq;
+	}
+
+	HandleSeq operator()(const contin_seq& c)
+	{
+		for (contin_t cell_value : c)
+			cellSeq.push_back(mk_real_cell_exec(cell_value, row_num,
+			                                    labels[cell_number++]));
+
+		return cellSeq;
+	}
+
+	template<typename T>
+	HandleSeq operator()(const T& t) const
+	{}
+
+	int cell_number = 0;
```

Comment:
It's a bit dangerous to have this here because if you reapply the visitor it will crash. I would suggest to move it in the operators.